### DISCUSSION TITLE
Which Method i can use?

### DISCUSSION THREAD

**Discussion by AshuraMajestic** [community/community#138867] [Programming Help]
### Body

I have a large dataset with over 2 lac data of train.csv and test.csv both of same size. The csv contains image links from which I want to extract text and do some operation. When I use kaggle kernel or other platform it stats downloading the images one by one. As it speed is so slow that I have to wait for more than 3 hours for just train data , then test data will also take same time. What should I do? I also tried using the Kaglle GPU but same problem.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SuGAr9807** [comment]
Same question and i tried to use chunk by chunk reading. It takes me approx 4+ hour for training the model.

**Comment by tpmst** [comment]
Hello @AshuraMajestic,

Handling a large dataset of image links for text extraction can be challenging, especially on platforms like Kaggle where resources like network bandwidth may be limited. Here are some strategies to help optimize the process and potentially speed up the downloading and processing of your images:

## Parallelize the Download Process
Instead of downloading images one by one, you can parallelize the download process to take advantage of multiple threads or processes. This can significantly reduce the overall time required to download all images.

Python Example with concurrent.futures:
```
import requests
from concurrent.futures import ThreadPoolExecutor

def download_image(url):
    response = requests.get(url)
    # Assuming you are extracting text here and saving it
    # extract_text(response.content)
    return response.content

urls = ["http://example.com/image1.jpg", "http://example.com/image2.jpg", ...]  # Your list of image URLs

with ThreadPoolExecutor(max_workers=20) as executor:
    results = executor.map(download_image, urls)
```

## Check for API usage
If the images are hosted on a platform that provides an API, check if the API allows batch requests or has a higher bandwidth limit than simple HTTP requests.

## Conclusion
By applying these strategies, you should be able to reduce the time required to download and process images for text extraction or do Training and downloading at the same time. An other way could be Muti-Processing the training and learning so it runs at the same time.
